test/battle/move_effect/toxic_spikes.c

```diff
@@ -208,3 +208,65 @@ SINGLE_BATTLE_TEST("Toxic Spikes are removed by Poison-types affected by Magnet
         NOT STATUS_ICON(opponent, poison: TRUE);
     }
 }
+
+SINGLE_BATTLE_TEST("Toxic Spikes inflicts poison on switch in after Pecha Berry's activation") // #3207
```

Comment:
This test is essentially the same as `Pokemon can be further poisoned with Toxic spikes after a status healing hold effect was previously used` in `test/battle/hold_effect/cure_status.c` except with the sides reversed.

Please move this test over there and edit them both to march with the following:
- `PARAMETRIZE` for Lum and Pecha berries.
- Messages for both.
- Same test name, except with a ` - Player` and ` - Opponent` suffix.
- Removing the issue number.